fhnw-ph-schreibberatung.csl

```diff
@@ -0,0 +1,476 @@
+      <if type="article-journal">
+        <choose>
+          <if variable="volume" match="none">
+            <text variable="issue" prefix=", Heft "/>         
+          </if>
+          <else>
+            <text variable="volume" prefix=" "/>
+            <text variable="issue" prefix="/"/>
+          </else>
+        </choose>
+      </if>
+       
+      <else-if type="legal_case">
+        <text variable="volume" prefix=", "/>
+        <text variable="container-title" prefix=" "/>
+        <text variable="page" prefix=" S. "/>
+      </else-if>
+
+      <else-if type="bill book graphic legal_case motion_picture report song chapter" match="any">
+        <group prefix=". " delimiter=". ">
+          <group>
+            <text term="volume" form="short" text-case="capitalize-first" suffix=". " strip-periods="true"/>
+            <number variable="volume" form="numeric"/>
+          </group>
+          <group>
+            <number variable="number-of-volumes" form="numeric"/>
+            <text term="volume" form="short" prefix=" " suffix="." plural="true" strip-periods="true"/>
+          </group>
+        </group>
+      </else-if>
+      
+    </choose>
+  </macro>
+
+  <macro name="locators-chapter">
+    <choose>
+      <if type="chapter paper-conference" match="any">
+        <group prefix=". ">
+          <!-- <text variable="volume" suffix="."/> -->
+          <text variable="page" prefix="S. "/>
+        </group>
+      </if>
+    </choose>
+  </macro>
+
+  <macro name="locators-article">
+    <choose>
+ 
+      <if type="article-newspaper">
+        <group prefix=", " delimiter=", ">
+          <group>
+            <text variable="edition" suffix=" "/>
+            <text term="edition" prefix=" "/>
+          </group>
+          <group>
+            <text term="section" form="short" suffix=". " strip-periods="true"/>
+            <text variable="section"/>
+          </group>
+        </group>
+        <text variable="page" prefix=". S. "/>
+      </if>
+
+      <else-if type="article-journal">
+        <text variable="page" prefix=". S. "/>
+      </else-if>
+
+    </choose>
+  </macro>
+
+  <macro name="point-locators">
+    <group>
+      <choose>
+        <if locator="page" match="none">
+          <label variable="locator" form="short" suffix=" "/>
+        </if>
+      </choose>
+      <text variable="locator"/>
+    </group>
+  </macro>
+
+  <macro name="container-prefix">
+    <text term="in" text-case="capitalize-first" suffix=":"/>
+  </macro>
+
+  <macro name="container-title">
+    <choose>
+      <if type="chapter paper-conference" match="any">
+        <text macro="container-prefix" suffix=" "/>
+        <text macro="container-contributors" suffix=": "/>
+      </if>
+    </choose>
+    <choose>
+      <if type="article-journal article-newspaper post post-weblog webpage entry entry-dictionary entry-encyclopedia">
+        <text macro="container-prefix" suffix=" "/>
```

Comment:
the way you have this, the container-prefix will render even if there is no container. I'd take the "in" out of the macro and enclose the two in a group--that way they will only be rendered when the variable actually exists.